yal/tests/states_dictionary.md

```diff
@@ -0,0 +1,5 @@
+| state_name                | accepts_user_input | added_to_flow_results_app | description                                                                       |
+|---------------------------|--------------------|---------------------------|-----------------------------------------------------------------------------------|
+| state_display_preferences |        TRUE        |            TRUE           | displays profile fields to user. User response is string of which field to update |
+| state_update_age          |        TRUE        |            TRUE           | asks user to enter their age. User response is int                                |
+| state_update_age_confirm  |        TRUE        |            TRUE           | asks user to confirm new age choice. User response is string "yes" or "no"        |
```

Comment:
I think this is sufficient to generate the flow results specification, as at the moment, we've got all the types set to `text`. So all we need is the identifier, `state_name`; `type` hardcoded as `text` which has no additional options; and the label, which we can set to the `description` here.

Ideally though, we would be able to specify the types as well, but doing that in a table might be a bit difficult.

eg. for `state_update_age_confirm`, a better type would be `select_one`, which has an additional required field `choices`, which we would want to set to `["yes","no"]`. 

For `state_update_age`, there's the `numeric` type, which has an optional field `range`, which we might want to set to `[1,100]`. Basically codifying what we have in the `description` field in this table.

https://floip.gitbook.io/flow-results-specification/specification#select_one